packages/typespec-test/test/batch_modular/generated/typespec-ts/src/index.ts

```diff
@@ -1,4 +1,380 @@
+  JobPreparationAndReleaseTaskExecutionInformation,
+  JobPreparationTaskExecutionInformation,
+  JobPreparationTaskState,
+  TaskContainerExecutionInformation,
+  TaskFailureInformation,
+  TaskExecutionResult,
+  JobReleaseTaskExecutionInformation,
+  JobReleaseTaskState,
+  TaskCountsResult,
+  TaskCounts,
+  TaskSlotCounts,
+  Certificate,
+  CertificateState,
+  DeleteCertificateError,
+  CertificateFormat,
+  CertificateListResult,
+  NodeFileListResult,
+  NodeFile,
+  FileProperties,
+  BatchJobSchedule,
+  JobScheduleState,
+  Schedule,
+  JobSpecification,
+  JobScheduleExecutionInformation,
+  RecentJob,
+  JobScheduleStatistics,
+  BatchJobScheduleListResult,
+  BatchTask,
+  ExitConditions,
+  ExitCodeMapping,
+  ExitOptions,
+  JobAction,
+  DependencyAction,
+  ExitCodeRangeMapping,
+  TaskState,
+  AffinityInformation,
+  TaskExecutionInformation,
+  ComputeNodeInformation,
+  MultiInstanceSettings,
+  TaskStatistics,
+  TaskDependencies,
+  TaskIdRange,
+  BatchTaskListResult,
+  TaskAddCollectionResult,
+  TaskAddResult,
+  TaskAddStatus,
+  BatchError,
+  ErrorMessage,
+  BatchErrorDetail,
+  BatchTaskCollection,
+  BatchTaskListSubtasksResult,
+  SubtaskInformation,
+  SubtaskState,
+  ComputeNodeUser,
+  NodeUpdateUserParameters,
+  ComputeNode,
+  ComputeNodeState,
+  SchedulingState,
+  TaskInformation,
+  StartTaskInformation,
+  StartTaskState,
+  ComputeNodeError,
+  ComputeNodeEndpointConfiguration,
+  InboundEndpoint,
+  NodeAgentInformation,
+  VirtualMachineInfo,
+  NodeRebootParameters,
+  ComputeNodeRebootOption,
+  NodeReimageParameters,
+  ComputeNodeReimageOption,
+  NodeDisableSchedulingParameters,
+  DisableComputeNodeSchedulingOption,
+  ComputeNodeGetRemoteLoginSettingsResult,
+  UploadBatchServiceLogsResult,
+  UploadBatchServiceLogsConfiguration,
+  ComputeNodeListResult,
+  NodeVMExtension,
+  VMExtensionInstanceView,
+  InstanceViewStatus,
+  StatusLevelTypes,
+  NodeVMExtensionList,
+} from "./api/models.js";
+export {
+  listSupportedImages,
+  listPoolNodeCounts,
+  AccountListSupportedImagesOptions,
+  AccountListPoolNodeCountsOptions,
+} from "./api/Account.js";
+export {
+  listApplications,
+  get,
+  ApplicationsListApplicationsOptions,
+  ApplicationsGetOptions,
+} from "./api/Applications.js";
+export { createBatchService } from "./api/BatchServiceContext.js";
+export {
+  addCertificate,
+  listCertificates,
+  cancelCertificateDeletion,
+  deleteCertificate,
+  getCertificate,
+  CertificatesAddCertificateOptions,
+  CertificatesListCertificatesOptions,
+  CertificatesCancelCertificateDeletionOptions,
+  CertificatesDeleteCertificateOptions,
+  CertificatesGetCertificateOptions,
+} from "./api/Certificates.js";
+export {
+  getComputeNodeExtensions,
+  listComputeNodeExtensions,
+  ComputeNodeExtensionsGetComputeNodeExtensionsOptions,
+  ComputeNodeExtensionsListComputeNodeExtensionsOptions,
+} from "./api/ComputeNodeExtensions.js";
+export {
+  addUser,
```

Comment:
How do you think if we have duplicated method names under diff folder?

Reply:
This is a good point, let's address it later on.
Maybe we can start by failing and show a meaningful error to the users asking to get some projectedNames in client.tsp

Reply:
Make sure we improve to faile the code gen not generated but compiler error.